src/sentry/api/endpoints/rule_snooze.py

```diff
@@ -0,0 +1,118 @@
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import audit_log, features
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectAlertRulePermission, ProjectEndpoint
+from sentry.api.serializers import Serializer, register, serialize
+from sentry.api.serializers.rest_framework.base import CamelSnakeSerializer
+from sentry.incidents.models import AlertRule
+from sentry.models import Rule, RuleSnooze
+
+
+class RuleSnoozeValidator(CamelSnakeSerializer):
+    target = serializers.CharField(required=True, allow_null=False)
+    rule = serializers.BooleanField(required=False, allow_null=True)
+    alert_rule = serializers.BooleanField(required=False, allow_null=True)
+    until = serializers.DateTimeField(required=False, allow_null=True)
+
+
+@register(RuleSnooze)
+class RuleSnoozeSerializer(Serializer):  # type: ignore
+    def serialize(self, obj, attrs, user, **kwargs):
+        result = {
+            "ownerId": obj.owner_id,
+            "userId": obj.user_id or "everyone",
+            "until": obj.until or "forever",
+            "dateAdded": obj.date_added,
+            "ruleId": obj.rule_id,
+            "alertRuleId": obj.alert_rule_id,
+        }
+        return result
+
+
+def can_edit_alert_rule(rule, organization, user):
+    rule_owner = rule.owner
+    if rule_owner:
+        if rule_owner.team not in rule_owner.team.objects.get_for_user(organization, user):
+            return False
+    return True
+
+
+@region_silo_endpoint
+class RuleSnoozeEndpoint(ProjectEndpoint):
+    permission_classes = (ProjectAlertRulePermission,)
+
+    def post(self, request: Request, project, rule_id) -> Response:
+        if not features.has("organizations:mute-alerts", project.organization, actor=None):
+            return Response(
+                {"detail": "This feature is not available for this organization."},
+                status=status.HTTP_401_UNAUTHORIZED,
+            )
+
+        serializer = RuleSnoozeValidator(data=request.data)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        data = serializer.validated_data
+
+        issue_alert_rule = None
+        metric_alert_rule = None
+
+        user_id = request.user.id if data.get("target") == "me" else None
+
+        if data.get("rule"):
+            try:
+                issue_alert_rule = Rule.objects.get(id=rule_id)
+            except Rule.DoesNotExist:
+                raise serializers.ValidationError("Rule does not exist")
+
+        if data.get("alert_rule"):
+            try:
+                metric_alert_rule = AlertRule.objects.get(id=rule_id)
+            except AlertRule.DoesNotExist:
+                raise serializers.ValidationError("Rule does not exist")
+
+        rule = issue_alert_rule or metric_alert_rule
+        if not can_edit_alert_rule(rule, project.organization, request.user):
```

Comment:
I think a user should be able to mute any alert rule for themselves (assuming they're a member of the org), regardless of who owns it.